libnavui-base/src/main/java/com/mapbox/navigation/ui/base/util/ViewUtils.kt

```diff
@@ -0,0 +1,58 @@
+package com.mapbox.navigation.ui.base.util
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.util.Base64
+import android.view.View
+import com.mapbox.navigation.core.MapboxNavigation
+import java.io.ByteArrayOutputStream
+import kotlin.math.min
+import kotlin.math.roundToInt
+
+/**
+ * Provides utilities to capture and encode screenshots, accepted by [MapboxNavigation.postUserFeedback].
+ */
+object ViewUtils {
+
+    /**
+     * Captures screenshot of a [View] and returns it as a [Bitmap].
+     * Warning: not suitable for Mapbox [MapView]. Use its [snapshot] function instead.
+     *
+     * @return captured [Bitmap]
+     */
+    @JvmStatic
+    fun View.captureBitmap(): Bitmap {
```

Comment:
Even this is exposed we're not accessing to it. Do we really need to expose this extension function? 🤔 

Also, I'm worried about the warning 

```kotlin
* Warning: not suitable for Mapbox [MapView]. Use its [snapshot] function instead.
```

BTW have you seen the implementation and APIs we expose in `v1.x` https://github.com/mapbox/mapbox-navigation-android/blob/release-v1.x/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/internal/utils/ViewUtils.java? Maybe we can do something similar so the warning won't be necessary.

Reply:
> suggested to have such a function to support users who don't use `MapView` from Mapbox Maps SDK. I would prefer not to expose it, because it will most likely not work for map views from another SDKs as well. 

Yeah, I'm with you. We shouldn't expose it, especially if we haven't heard from any customer (we can always add it in the future without breaking SemVer otherwise).

> I was actually using the v1.x implementation as a reference and I already verified that it doesn't work for `MapView`.

I'd swear that used to work in the past. Did something change downstream with the latest version of Maps v10? 🤔 
In any case, if it isn't working even more reasons to not expose it 👍

Reply:
> I'd swear that used to work in the past. Did something change downstream with the latest version of Maps v10? 🤔 In any case, if it isn't working even more reasons to not expose it 👍

I checked again and found out that the old `ViewUtils.captureView` function actually works, but only if you call `MapView.snapshot` beforehand. Otherwise the map is just black. I'll try to find a simpler way to capture views.

Reply:
I was wrong, previous implementation works in 1tap only with the help of `ImageView`, which is shown above the `MapView`. So in 1tap we have a pretty sophisticated process: we grab a current snapshot using `MapView.snapshot` function, then we display it in an `ImageView` above the map, and finally we capture a screenshot using `ViewUtils.captureView` function. This helps us have a screenshot with the map and all the views we have above it, like trip progress and banner instructions. `MapView.snapshot` provides a screenshot, which only has a map, including a puck and a route line.

Reply:
Alright, I implemented this workaround in `FeedbackActivity`. Also mentioned in `ViewUtils` docs that the user has to capture `MapView` snapshots separately.

Reply:
Should we accept `View` and check for `is instance of MapView` and do it from our side? 🤔

Want to remove as much complications as possible from client side. As we know, relying on docs is complicated 😅

Reply:
Found a way to do everything in one function. The idea is to set the snapshot as a [foreground](https://developer.android.com/reference/android/view/View#setForeground(android.graphics.drawable.Drawable)) for the `MapView`, that allows capturing the view tree as usual.

Reply:
Yeah, way to go 🚀